File: lib/algorithms/selection_sort.js
function selectionSort(array) {
  for(var i = 0; i < array.length; i++) {
    var lowestNumIndex = i;
    for(var j = i+1; j < array.length; j++) {
      if(array[lowestNumIndex] > array[j]) {
        lowestNumIndex = j;
      }
    }

    if(lowestNumIndex !== i) {
      var temp = array[i]
      array[i] = array[lowestNumIndex]
      array[lowestNumIndex] = temp;
    }
  }
  return array;
}

array = [3,2,7,6,4]
console.log(selectionSort(array))

array = []
console.log(selectionSort(array))

File: lib/algorithms/selection_sort.rb
def selection_sort(arr)
  n = arr.length - 1
  n.times do |i|
    min_index = i
    (i + 1).upto(n) do |j|
      min_index = j if arr[j] < arr[min_index]
    end
    arr[i], arr[min_index] = arr[min_index], arr[i] if min_index != i
    p arr
  end
  arr
end

arr = [3,2,8,5,7,6]
pp selection_sort(arr)